core/src/main/java/org/apache/druid/java/util/common/concurrent/ScheduledExecutors.java

```diff
@@ -53,7 +53,11 @@ public static void scheduleWithFixedDelay(
           public Signal call()
           {
             runnable.run(); // (Exceptions are handled for us)
-            return Signal.REPEAT;
+            if (exec.isShutdown()) {
+              return Signal.STOP;
```

Comment:
please add a log line here. It will be hard to debug if someone wrongly shuts down the exec.

Reply:
Done. By the way it seems that I can't passed the 'continuous-integration/travis-ci/pr'. Because existing test code seems to never cover ScheduledExecutors which is a util class. Orz...

Reply:
@nishantmonu51 Hi nishantmonu51, sorry to bother you. I have tested this change in our Druid cluster. What should I do next ?

Reply:
nit: I would make that a `log.warn(..)` because debug would hardly be enabled and this log is not gonna pollute the logs, but looks good otherwise.

thanks.

Reply:
@himanshug Thanks for your review. I have changed the log level to WARN.